src/Lykke.Job.TransactionHandler/Queues/TransferQueue.cs

```diff
@@ -158,18 +183,97 @@ public void Stop()
+                    case OperationType.TransferToTrusted:
+                        return await ProcessEthTransferTrustedWallet(ethTxRequest, TransferType.ToTrustedWallet);
+                    case OperationType.TransferFromTrusted:
+                        return await ProcessEthTransferTrustedWallet(ethTxRequest, TransferType.FromTrustedWallet);
+                    default:
+                        return true;
+                }
+            }
 
+            return true;
+        }
+
+        private async Task<bool> ProcessEthTransferTrustedWallet(IEthereumTransactionRequest txRequest, TransferType transferType)
+        {
+            var ethError = string.Empty;
+
+            try
+            {
+                var asset = await _assetsService.TryGetAssetAsync(txRequest.AssetId);
+                var clientAddress = await _bcnClientCredentialsRepository.GetClientAddress(txRequest.ClientId);
+                var hotWalletAddress = _settings.HotwalletAddress;
 
+                string addressFrom;
+                string addressTo;
+                Guid transferId;
+                string sign;
+                switch (transferType)
+                {
+                    case TransferType.ToTrustedWallet:
+                        addressFrom = clientAddress;
+                        addressTo = hotWalletAddress;
+                        transferId = txRequest.SignedTransfer.Id;
+                        sign = txRequest.SignedTransfer.Sign;
+                        break;
+                    case TransferType.FromTrustedWallet:
+                        addressFrom = hotWalletAddress;
+                        addressTo = clientAddress;
+                        transferId = txRequest.Id;
+                        sign = string.Empty;
+                        break;
+                    default:
+                        await _log.WriteErrorAsync(nameof(TransferQueue), nameof(ProcessEthTransferTrustedWallet),
+                            "Unknown transfer type", null);
+                        return false;
+                }
+                var ethResponse = await _srvEthereumHelper.SendTransferAsync(transferId, sign, asset, addressFrom,
+                    addressTo, txRequest.Volume);
+
+                if (ethResponse.HasError)
+                {
+                    ethError = ethResponse.Error.ToJson();
+                    await _log.WriteErrorAsync(nameof(TransferQueue), nameof(ProcessEthTransferTrustedWallet), ethError, null);
+
+                    return false;
+                }
+            }
+            catch (Exception e)
+            {
+                await _log.WriteErrorAsync(nameof(TransferQueue), nameof(ProcessEthTransferTrustedWallet), e.Message, e);
+                ethError = $"{e.GetType()}\n{e.Message}";
+            }
+
+            if (!string.IsNullOrEmpty(ethError))
+            {
+                await _log.WriteErrorAsync(nameof(TransferQueue), nameof(ProcessEthTransferTrustedWallet), ethError,
```

Comment:
This is a repeating of line 266. The only way we got into this 'if' section if there was an exception a few lines before.

Reply:
Correct, here is some strange error handling logic, fixed.